src/backend/api/handlers/search.py

```diff
@@ -0,0 +1,53 @@
+from flask import Response
+
+from backend.api.handlers.decorators import api_authenticated
+from backend.api.handlers.helpers.model_properties import (
+    filter_event_properties,
+    filter_team_properties,
+    ModelType,
+)
+from backend.api.handlers.helpers.profiled_jsonify import profiled_jsonify
+from backend.api.handlers.helpers.track_call import track_call_after_response
+from backend.common.consts.api_version import ApiMajorVersion
+from backend.common.decorators import cached_public
+from backend.common.helpers.season_helper import SeasonHelper
+from backend.common.models.team import Team
+from backend.common.queries.event_query import EventListQuery
+from backend.common.queries.team_query import TeamListQuery
+
+
+@api_authenticated
+@cached_public
```

Comment:
Add a TODO to bump the cache time on this up to a day. It's fine to keep this low while we are testing.